Tooltip.jsx

```diff
@@ -3,178 +3,152 @@
+import TooltipTrigger from "react-popper-tooltip";
+
+import colors from "./colors";
+import {zIndices} from "./tools/zIndices";
+
+const verticalOffset = 8;
+
+type Props = {|
+  +placement?: "top" | "bottom" | "left" | "right",
+  +overlay: React.Node,
+  +trigger?: "hover" | "click",
+  +mouseEnterDelay?: number, // seconds
+  +mouseExitDelay?: number, // seconds
+  +maxWidth?: number,
+  +children: string | React.Node,
+  +triggerClassName?: string,
+|};
+
+export default function Tooltip({
+  placement = "right",
+  overlay,
+  trigger = "hover",
+  children,
+  mouseEnterDelay = 0,
+  mouseExitDelay = 0,
+  maxWidth = null,
+  triggerClassName = "",
+}: Props) {
+  return (
+    <TooltipTrigger
+      placement={placement}
+      trigger={trigger}
+      tooltip={MakeContainer(overlay, maxWidth)}
+      delayShow={mouseEnterDelay * 1000}
+      delayHide={mouseExitDelay * 1000}
+      modifiers={{
+        offset: {
+          offset: `0,${verticalOffset}px`,
+        },
+      }}
+    >
+      {MakeTrigger(children, triggerClassName)}
+    </TooltipTrigger>
+  );
 }
-let waitForRcTooltip = initModuleState();
 
-/**
- * Reset module state. Only useful during testing.
- */
-export function insideTestResetModuleState() {
-  RcTooltip = null;
-  waitForRcTooltipResolve = null;
-  waitForRcTooltipReject = null;
-  waitForRcTooltip = initModuleState();
+function MakeContainer(children: React.Node, maxWidth: ?number) {
+  return ({
+    arrowRef,
+    tooltipRef,
+    getArrowProps,
+    getTooltipProps,
+    placement,
+  }) => (
+    <div
+      {...getTooltipProps({
+        ref: tooltipRef,
+        className: css(styles.tooltipContainer),
+        style: maxWidth != null ? {maxWidth} : {},
+      })}
+    >
+      <div
+        {...getArrowProps({
+          ref: arrowRef,
+          className: css(styles.tooltipArrow, getPlacementStyles(placement)),
+        })}
+      />
+      <div className={css(styles.tooltipContent)}>{children}</div>
+    </div>
+  );
 }
 
-/**
- * Load rc-tooltip, then store a reference to the code via a side-effect, and
- * resolve the promise so that all components can re-render.
- */
-function beginLoadingRcTooltip() {
-  // disabling dynamic-import-webchunknamme here to bundle both assets together
-  /* eslint-disable flexport/dynamic-import-webchunkname */
-  const rcPromise = import(/* webpackChunkName: "rc-tooltip" */ "./BetterRcTooltip");
-  // $FlowFixMe(uforic): Stylesheets don't have types
-  const styleSheetPromise = import(/* webpackChunkName: "rc-tooltip" */ "vendor_stylesheets/rc-tooltip.css");
-  return Promise.all([rcPromise, styleSheetPromise]).then(
-    ([{default: RcTooltip}: {default: RcTooltipT}]) => {
-      invariant(
-        waitForRcTooltipResolve !== null,
-        "Resolve function must be present"
-      );
-      waitForRcTooltipResolve(RcTooltip);
-    },
-    waitForRcTooltipReject
+function MakeTrigger(children: string | React.Node, className: string) {
```

Comment:
This isn't a React component so it shouldn't begin with a capital letter.

If you want it to be a component, call it like `<MakeTrigger className={...} />` instead. I think this is probably the right move.

```suggestion
function makeTrigger(children: string | React.Node, className: string) {
```